### PR TITLE
Refactor products_controller move #find_order to before_action and ad…

### PR DESCRIPTION
**Pull request by tanmay3011**
…d specs for find order

### CODE REVIEW

File: api/spec/controllers/spree/api/v1/checkouts_controller_spec.rb [hunk 1]
```diff
@@ -241,7 +241,7 @@ def controller.load_order(*)
       end
 
       it "can transition from confirm to complete" do
-        order.update_columns(completed_at: Time.current, state: 'complete')
+        order.update_columns(state: 'confirm')
```
Comment (jitendra) [inline api/spec/controllers/spree/api/v1/checkouts_controller_spec.rb:244]: update_column

**Comment by tanmay3011** [comment]
@jitendra  Please review now... Earlier the branch wasn't rebased

**Comment by tanmay3011** [comment]
`find_by_id_or_slug` cannot be used since friendly_id can find using any query field

Definition of find used in friendly_id 

```
 def find(*args)
    id = args.first
    return super if args.count != 1 || id.unfriendly_id?
    first_by_friendly_id(id).tap {|result| return result unless result.nil?}
    return super if potential_primary_key?(id)
    raise ActiveRecord::RecordNotFound, "can't find record with friendly id: #{id.inspect}"
  end

 private
 def first_by_friendly_id(id)
   find_by(friendly_id_config.query_field => id)
 end
```

**Comment by jitendra** [comment]
@tanmay3011 thanks for the definition. Please go ahead.

**Pull request closed without merging**